### ISSUE TITLE
Optimize snap-to-grid code

### ISSUE THREAD

**Issue by ehlers** [GNS3/gns3-gui#1599]
GNS3 version 1.5.3dev1 on Darwin (64-bit) with Python 3.5.2 Qt 5.7.0.

After reading the QT documentation http://doc.qt.io/qt-5.6/qgraphicsitem.html#GraphicsItemChange-enum I'm quite certain that the current snap-to-grid implementation can be optimized.

Currently, when snap-to-grid is enabled, a position change of an item results in this flow of events:
- an ItemPositionChange event with the new position, which is ignored by GNS3
- the item is moved to the new position
- in the ItemPositionHasChanged event GNS3 calculates the nearest grid position and issues a setpos() to the grid position
- an ItemPositionChange event with the grid position, ignored by GNS3
- the item is moved to the grid position
- in the ItemPositionHasChanged event the current position is equal to the grid position, so nothing is done

This results in sporadic jitter, when a router is constantly moved a bit, but not far enough to reach the next grid position, here a video:
![snap_to_grid](https://cloud.githubusercontent.com/assets/896406/19622722/c2a88c9a-98b0-11e6-8b6e-ca5f21fd1478.gif)

My suggestion is to do the snap-to-grid functionality in the ItemPositionChange event. It just has to return the nearest grid position (when snap-to-grid is enabled).

This results in this flow of events, when the position changes.
- an ItemPositionChange event with the new position, GNS3 calculates the nearest grid position and returns it.
- the item is moved to the grid position
- the ItemPositionHasChanged event is ignored.

This results in only half the number of events and the jitter from above is completely gone.

I don't suggest to add this patch into stable V1.5, but it might be interesting for V2.0:
```
diff --git a/gns3/items/node_item.py b/gns3/items/node_item.py
index e7be6ac..bbcc486 100644
--- a/gns3/items/node_item.py
+++ b/gns3/items/node_item.py
@@ -387,14 +387,12 @@ class NodeItem():
         :param value: value of the change
         """

-        if change == QtWidgets.QGraphicsItem.ItemPositionHasChanged and self.isActive() and self._main_window.uiSnapToGridAction.isChecked():
+        if change == QtWidgets.QGraphicsItem.ItemPositionChange and self.isActive() and self._main_window.uiSnapToGridAction.isChecked():
             GRID_SIZE = 75
             mid_x = self.boundingRect().width() / 2
-            tmp_x = (GRID_SIZE * round((self.x() + mid_x) / GRID_SIZE)) - mid_x
+            value.setX((GRID_SIZE * round((value.x()+mid_x)/GRID_SIZE)) - mid_x)
             mid_y = self.boundingRect().height() / 2
-            tmp_y = (GRID_SIZE * round((self.y() + mid_y) / GRID_SIZE)) - mid_y
-            if tmp_x != self.x() and tmp_y != self.y():
-                self.setPos(tmp_x, tmp_y)
+            value.setY((GRID_SIZE * round((value.y()+mid_y)/GRID_SIZE)) - mid_y)

         # dynamically change the renderer when this node item is selected/unselected.
         if change == QtWidgets.QGraphicsItem.ItemSelectedChange:
```

**Comment by noplay** [comment]
I agree it seem better

**Comment by noplay** [comment]
Merged in 2.0
